src/main/java/carpetaddonsnotfound/mixins/FlowerPotBlock_PickFlowersFromPots.java

```diff
@@ -0,0 +1,57 @@
+package carpetaddonsnotfound.mixins;
+
+import carpetaddonsnotfound.CarpetAddonsNotFoundSettings;
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockState;
+import net.minecraft.block.Blocks;
+import net.minecraft.block.FlowerPotBlock;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ActionResult;
+import net.minecraft.util.Hand;
+import net.minecraft.util.hit.BlockHitResult;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.World;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(FlowerPotBlock.class)
+abstract class FlowerPotBlock_PickFlowersFromPots {
+  @Inject(
+          method = "onUse",
+          at = @At(
+                  value = "TAIL"
+          ),
+          cancellable = true
+  )
+  private void onUse(BlockState state,
+                     World world,
+                     BlockPos pos,
+                     PlayerEntity player,
+                     Hand hand,
+                     BlockHitResult hit,
+                     CallbackInfoReturnable<ActionResult> cir) {
+    Block potBlock = state.getBlock();
+    Block content;
+    if (!CarpetAddonsNotFoundSettings.pickFlowersFromPots || !(potBlock instanceof FlowerPotBlock) ||
```

Comment:
I think you can just do:
```java
... !(potBlock instanceof FlowerPotBlock content) || content.getContent() == Blocks.AIR ...
```

Which will set `content` to `(FlowerPotBlock) potBlock` if `potBlock` is an instance of `FlowerPotBlock`.